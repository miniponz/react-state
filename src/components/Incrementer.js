import React, { PureComponent } from 'react';

export default class Incrementer extends PureComponent {
  state = {
    count: 0
  }

  increment = () => {
    this.setState(state => {
      return { count: state.count + 1 };
    });
  }

  render() {
    let { count } = this.state; 

    return (
      <>
        <p>Count: {count}</p>
        <button onClick={this.increment}>Increment Here!</button>
      </>
    );
  }

}
