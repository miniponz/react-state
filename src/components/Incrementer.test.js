import React from 'react';
import { shallow } from 'enzyme';
import Incrementer from './Incrementer';

describe('incrementer test', () => {
  it('renders incrementer', () => {
    const wrapper = shallow(<Incrementer />);
    expect(wrapper).toMatchSnapshot();
  });

  it('increments count on button click', () => {
    const wrapper = shallow(<Incrementer />);
    wrapper.find('button').at(0).simulate('click');

    expect(wrapper.state('count')).toEqual(1);
  });
});
