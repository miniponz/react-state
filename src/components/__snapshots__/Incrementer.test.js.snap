// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`incrementer test renders incrementer 1`] = `
<Fragment>
  <p>
    Count: 
    0
  </p>
  <button
    onClick={[Function]}
  >
    Increment Here!
  </button>
</Fragment>
`;
